services/core/MongodbHistorian/mongodb/historian.py

```diff
@@ -345,10 +359,12 @@ def periodic_rollup(self):
                 self.DAILY_COLLECTION, bulk_publish_day, day_ids, db)
 
     def get_last_updated_data(self, db, collection):
+        id = ""
         cursor = db[collection].find({}).sort(
             "last_updated_data", pymongo.DESCENDING).limit(1)
         for row in cursor:
-            return row['last_updated_data']
+            id = row.get('last_updated_data')
```

Comment:
Though I think this will do what you want because you are using limit(1) isn't there a find_single or find_one that you can use without having to deal with the loop here?